packages/teams-js/src/public/search.ts

```diff
@@ -0,0 +1,131 @@
+import { sendMessageToParent } from '../internal/communication';
+import { registerHandler, removeHandler } from '../internal/handlers';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { runtime } from './runtime';
+
+/**
+ * Allows you to interact with the host search experience
+ *
+ * @beta
+ */
+export namespace search {
+  const onChangeHandlerName = 'search.queryChange';
+  const onClosedHandlerName = 'search.queryClosed';
+  const onExecutedHandlerName = 'search.queryExecute';
+
+  /**
+   * This interface contains information pertaining to the search term in the host search experience
+   */
+  export interface SearchQuery {
+    /** The current search term in the host search experience */
+    searchTerm: string;
+
+    // TODO: may need some sort of timestamp or sequence value to
+    // ensure messages are processed in correct order / combine them
+    // having any sort of logic around combining messages or sorting them
+    // would make sense to go into the teamsjs-sdk layer
+    // timestamp: number;
+  }
+
+  export type SearchQueryHandler = (query: SearchQuery) => void;
+
+  /**
+   * Allows the caller to register for various events fired by the host search experience.
+   * Calling this function will cause the host search experience to set its default scope to
+   * the name of your application.
+   * 
+   * Your application should *not* re-render inside of these callbacks, there may be a large number
+   * of onChangeHandler calls if the user is changing the searchQuery rapidly.
+   *
+   * @param onChangeHandler - This handler will be called when the user begins searching and every
+   * time the user changes the contents of the query. The value of the query is the current term
+   * the user is searching for. Should be used to put your application into whatever state is used
+   * to handle searching. This handler will be called once with an empty {@link SearchQuery.searchTerm}
+   * when search is beginning. 
+   * @param onClosedHandler - This handler will be called when the user finishes searching. Should be
```

Comment:
What does this mean to "finish searching"? Focus away from the searchbox? Some explicit "X" UI on the host part to 'finish' a search? I think this could use some more detail.

Reply:
Agreed, but I assume to some extent this will be host-specific as the UI around search may vary from host to host. In Teams for example, you can either press the 'X' button in the search bar or press Esc a few times. Definitely agree that part of removing the beta tag here would be being as crisp as possible about exactly what this means

Reply:
why do we need to know what the last query was before ending search?

Reply:
Ultimately, I think how that works is something I'm open to you and your team changing. My thinking was that apps may want to be able to access what the last term was in the search box for a few different reasons (their own telemetry, putting the term back into the search box in the future, etc.)